### PR TITLE
Bugs: Allow non-integer fee values, UPDATE endpoint fixes

### PR DESCRIPTION
**Pull request by e-wai**
## Implementation description
* Ages and capacities should be integers, but fees can be floats?
* See comments

## Steps to test
1. Try UPDATE endpoint with floats
2. Try POST and PATCH camp endpoints with invalid JSONs

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/campValidators.ts [hunk 1]
```diff
@@ -168,29 +165,34 @@ export const updateCampDtoValidator = async (
...
-  const body = JSON.parse(req.body.data);
+  let body;
+  try {
+    body = JSON.parse(req.body.data);
+  } catch (e: unknown) {
+    return res.status(400).send(getErrorMessage(e));
+  }
+  if (!body) {
+    return res.status(400).send("JSON parsing failed");
+  }
```
Comment (e-wai) [inline backend/typescript/middlewares/validators/campValidators.ts:176]: same as create

File: backend/typescript/middlewares/validators/campValidators.ts [hunk 2]
```diff
@@ -168,29 +165,34 @@ export const updateCampDtoValidator = async (
...
   if (!validatePrimitive(body.name, "string")) {
     return res.status(400).send(getApiValidationError("name", "string"));
   }
-  if (body.description && !validatePrimitive(body.description, "string")) {
+  if (!validatePrimitive(body.description, "string")) {
```
Comment (e-wai) [inline backend/typescript/middlewares/validators/campValidators.ts:180]: redundant, the `validatePrimitive` method checks against `null` and `undefined`?

File: backend/typescript/services/implementations/campService.ts [hunk 3]
```diff
@@ -389,9 +389,9 @@ class CampService implements ICampService {
         oldCamp.active &&
         (camp.fee !== oldCamp.fee ||
           camp.dropoffFee !== oldCamp.dropoffFee ||
-          camp.pickupFee !== oldCamp.dropoffFee)
+          camp.pickupFee !== oldCamp.pickupFee)
```
Comment (e-wai) [inline backend/typescript/services/implementations/campService.ts:392]: This took me a ridiculous amount of time to find. I kept just looking at it.

**Review by dhmi19 — APPROVED** [review]
good catch on that bug! just make sure we weren't using integer for a reason

**Pull request merged**